### PR TITLE
PUB-1077 Complete E2E Journey

### PR DESCRIPTION
**Pull request by commitBlob**
### JIRA link ###
https://tools.hmcts.net/jira/browse/PUB-1077

### Change description ###
- refactor e2e public and private journeys
- add missing e2e steps
- add e2e for both signed in and public banner navigations

**Notes:**
- live case journey is currently disabled (to be re-introduced in the future)
- add subscription by URN is disabled as there are no cases with URN in the system at the moment

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/test/e2e/tests/specs.test.ts [hunk 1]
```diff
@@ -275,62 +228,64 @@ if (process.env.EXCLUDE_E2E === 'true') {
...
-}
-if (process.env.EXCLUDE_E2E === 'true') {
-  describe('Verified user', () => {
-    describe('Sign In Page', () => {
-      const HMCTSAccountUrl = 'https://hmcts-sjp.herokuapp.com/sign-in-idam.html';
+});
+
+describe('Verified user', () => {
+  describe('Sign In Page', () => {
+    const HMCTSAccountUrl = 'https://hmcts-sjp.herokuapp.com/sign-in-idam.html';
```
Comment (kurt-uk) [inline src/test/e2e/tests/specs.test.ts:235]: is this still needed? is this not from stitching?

File: src/test/e2e/tests/specs.test.ts [hunk 2]
```diff
@@ -275,62 +228,64 @@ if (process.env.EXCLUDE_E2E === 'true') {
...
-          await signInPage.enterText(config.get('secrets.pip-ss-kv.B2C_USERNAME'), 'EmailField');
-          await signInPage.enterText(config.get('secrets.pip-ss-kv.B2C_PASSWORD'), 'PasswordField');
-          accountHomePage = await signInPage.clickSignIn();
-          await browser.pause(2000);
-        });
+      it('should select \'Sign in with my P&I details\' option, navigate to the login page, and sign in', async () => {
+        await signInPage.open('/sign-in');
+        await signInPage.selectOption('SignInRadio3');
+        await signInPage.clickContinueForRadio3();
+        console.log('B2C_USERNAME', process.env.B2C_USERNAME);
```
Comment (kurt-uk) [inline src/test/e2e/tests/specs.test.ts:263]: console log left in

File: src/test/e2e/tests/specs.test.ts [hunk 3]
```diff
@@ -275,62 +228,64 @@ if (process.env.EXCLUDE_E2E === 'true') {
...
+        await signInPage.enterText(process.env.B2C_USERNAME, 'EmailField');
+        await signInPage.enterText(process.env.B2C_PASSWORD, 'PasswordField');
+        accountHomePage = await signInPage.clickSignIn();
+        await browser.pause(2000);
```
Comment (kurt-uk) [inline src/test/e2e/tests/specs.test.ts:267]: pause left in

**Review by kurt-uk — CHANGES_REQUESTED** [review]
looks good! just a couple qs

File: src/test/e2e/tests/specs.test.ts (hunk 2 above)
Comment (commitBlob) [reply src/test/e2e/tests/specs.test.ts:263]: this one is on purpose so we can see if there are any issues on the pipeline

File: src/test/e2e/tests/specs.test.ts (hunk 3 above)
Comment (commitBlob) [reply src/test/e2e/tests/specs.test.ts:267]: also on the purpose as we need to wait b2c to sign user in, sometimes takes longer and sometimes is snappy

File: src/test/e2e/tests/specs.test.ts (hunk 1 above)
Comment (commitBlob) [reply src/test/e2e/tests/specs.test.ts]: After talking with Alberto we decided to remove 2 disabled steps

**Review by alberto-hortiguela — DISMISSED** [review]

**Review by kurt-uk — DISMISSED** [review]

**Review by alberto-hortiguela — APPROVED** [review]

**Review by furnivall — APPROVED** [review]
lgtm :)

**Pull request merged**